examples/llm_serving/model/bloom_model.py

```diff
@@ -253,7 +255,7 @@ def __call__(self, x):
 
 class FlaxBloomMLP(nn.Module):
     config: BloomConfig
-    dtype: jnp.dtype = jnp.float16
+    dtype: jnp.dtype = jnp.float32
```

Comment:
Why are all the dtype modifications needed in this file? Please explain. Otherwise, please revert them.

Reply:
These are changed to maximally match the hf flax bloom model, and to match the logits on each step to the same accuracy. I'll try to change these to fp16 and check if the output changes.

Reply:
The one in FlaxBloomMLP has been changed to float16.
The one in FlaxBloomForCausalLMModule.setup()'s lm_head is kept.
Below are two runs, top one with FlaxBloomForCausalLMModule.setup()'s lm_head set to float16 and the bottom one is float32:
<img width="1146" alt="Screen Shot 2022-10-03 at 09 51 16" src="https://user-images.githubusercontent.com/65443542/193594490-34a54eff-95f6-4c5b-86bd-0ac88e730bf6.png">
The outputs are different, so this float32 to be kept.